libqfieldsync/utils/file_utils.py

```diff
@@ -216,3 +226,90 @@ def is_valid_filepath(path: str) -> bool:
             return False
 
     return True
+
+
+def update_symbols_path_assets(symbol, new_path: Path) -> None:
+    """
+    Updates the path of a symbol layer.
+    Args:
+        symbol: The QGIS symbol (from a renderer).
+        new_path: The base path to which asset paths should be made relative to.
+    """
+    if symbol is None:
+        return
+
+    destination_dir = new_path / "assets"
+    destination_dir.mkdir(parents=True, exist_ok=True)
+
+    for symbol_layer in symbol.symbolLayers():
+        if isinstance(
+            symbol_layer, (QgsSvgMarkerSymbolLayer, QgsRasterMarkerSymbolLayer)
+        ):
+            source_path = Path(symbol_layer.path())
+
+            if not source_path.is_relative_to(new_path):
+                if source_path.exists():
+                    destination_path_file = destination_dir / source_path.name
+
+                    if not destination_path_file.exists():
+                        shutil.copy2(source_path, destination_path_file)
+
+                    symbol_layer.setPath(
+                        str(destination_path_file.relative_to(new_path))
+                    )
```

Comment:
```suggestion
        # filter out only symbology that includes SVG and skip the rest
        if not isinstance(
            symbol_layer, (QgsSvgMarkerSymbolLayer, QgsRasterMarkerSymbolLayer)
        ):
            continue

        source_path = Path(symbol_layer.path())

        # the symbol is already broken, it's file is not reachable
        if source_path.is_file():
            continue

        # if the symbol's path is already relative, we have nothing to do
        if source_path.is_relative_to(new_path):
            continue

        destination_path_file = destination_dir / source_path.name

        if not destination_path_file.exists():
            shutil.copy2(source_path, destination_path_file)

        symbol_layer.setPath(
            str(destination_path_file.relative_to(new_path))
        )
```
I hope it is a bit morer readable than 8 levels of nesting.

Reply:
I think the whole codebase is prefering the `joinpath` over `/` operator.

Reply:
What I was wondering is since we do a `shutil.copy2`, there might be something in `assets` already. Is there any chance to embed the SVG symbol dynamically, or it is way too complicated?

Reply:
Great!... Indeed is more readable!.... Thanks @suricactus!